### PR TITLE
refactor(`zetaclient`): use keyring backend file for hotkey

### PR DESCRIPTION
**Pull request by lumtis**
# Description

- Remove the password from the ZetaClient config 
- Add option `KeyringBackend` in ZetaClient config for the keyring backend used for the key
- Add the File keyring backend for hot key for encryption
- Use Keyring Backend file in the smoketests

When using keyring backend file, the password must be provided through the env var `HOTKEY_PASSWORD`

```
export HOTKEY_PASSWORD=foobarfoo
zetaclientd ... --keyring-backend file
```

Closes: https://github.com/zeta-chain/node/issues/1278

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

File: zetaclient/keys.go
Comment (CryptoFewka) [inline zetaclient/keys.go:196]: I strongly encourage us to require a password that is longer than 8 characters, especially since it is something that is going to be almost exclusively programmatically used from an environment variable. A longer passphrase will provide more entropy than an exceptionally complex, but only eight character long password. 

I feel like in this instance we should encourage this better practice, even though ultimately the password will get scraped from the environment variable by an attacker. 

Actual protection for the secret likely would only come from a hardware security module.

Comment (lumtis) [reply zetaclient/keys.go:196]: The 8 characters is the requirement for the Cosmos keyring, this is why I put this specific check.

I agree that HSM would be the optimal security model in practice. This PR is mainly to introduce the option to use a different keyring backend than test, we plan to add the HSM feature on top of it.

File: .github/workflows/build.yml
Comment (lumtis) [reply .github/workflows/build.yml:128]: There is a bug where the ZetaClient waits indefinitely without using this sleep. I can provide more log. There might be an eventual issue to fix that I think could be fixed in another PR

Comment (CryptoFewka) [reply .github/workflows/build.yml:128]: OK sweet. Yea an indefinite wait is far more pricey than 10 seconds =)

File: zetaclient/keys.go
Comment (CryptoFewka) [reply zetaclient/keys.go:196]: 8 characters is probably fine, and if we're matching a Cosmos keyring requirement then perhaps it shouldn't differ

**Review by kingpinXD — APPROVED** [review]

File: cmd/zetacored/get_pubkey.go
Comment (lumtis) [reply cmd/zetacored/get_pubkey.go:24]: You can provide spaces in the argument, single or double quotes `get-pubkey myTssKey "This is a very secure passphrase."` also works.

File: cmd/zetacored/get_pubkey.go
Comment (lumtis) [reply cmd/zetacored/get_pubkey.go]: 3d2cb02f

File: .github/workflows/build.yml
Comment (lumtis) [reply .github/workflows/build.yml]: I removed the 10 seconds sleep, and somehow the issue is no longer here. It may have been related to Github or somehow I introduced a fix at some point.

File: zetaclient/keys.go
Comment (brewmaster012) [inline zetaclient/keys.go:198]: As further improvement, i think storing the password in process memory is better than storing it in environment variable. The reason is that envivronment variable are in known places in a process (or maybe just the control block of a process?). I.e., I can on linux simply find the all environment variables in a running process: by
`$ cat /proc/2328594/environ `

If the password is stored in a regular process variable it will be much harder to find even if a shell is obtained by an attacker on that node that runs `zetaclientd`. 

This is not blocking this PR though so feel free to comment and resolve.

**Review by brewmaster012 — APPROVED** [review]

File: zetaclient/keys.go
Comment (lumtis) [reply zetaclient/keys.go:198]: It could also be added as an additional option as well.
The `KeyringBackend` options we define in ZetaClient don't need to match exactly the `KeyringBackend` options of Cosmos (maybe should rename the option then).
We could have for example `KeyringBackendFilePasswordEnv` and `KeyringBackendFilePasswordMemory`.
I may think env var option can be convenient for teams that want to run a ZetaClient with a fullly Infrastructure as Code design.

Comment (CryptoFewka) [reply zetaclient/keys.go:198]: I agree that ENV should certainly be an option, but when we publish documentation we really should educate folks on the safety of their keys. Perhaps we should even go as far as to present a ranked list of options or flowchart.

I am still encountering people who think their keys/cryptocurrency is safe on a hardware wallet and therefore do not need to protect/worry about the mnemonic's security.

File: .github/workflows/build.yml
Comment (lumtis) [reply .github/workflows/build.yml:128]: Added back the 10 seconds as it was failing again... Created an issue for this.

**Comment by lumtis** [comment]
I will wait on a review from @kevinssgh before merging if you can have a look sometime. Thanks

**Review by kevinssgh — APPROVED** [review]

File: zetaclient/keys.go
Comment (lumtis) [reply zetaclient/keys.go:198]: https://github.com/zeta-chain/node/issues/1368

**Review by CryptoFewka — APPROVED** [review]

**Pull request merged**
